plutus-ledger-api/src/Plutus/V1/Ledger/Scripts.hs

```diff
@@ -157,23 +159,23 @@ scriptSize (Script s) = UPLC.programSize s
 -- See Note [Normalized types in Scripts]
 -- | Turn a 'CompiledCode' (usually produced by 'compile') into a 'Script' for use with this package.
 fromCompiledCode :: CompiledCode a -> Script
-fromCompiledCode = fromPlc . getPlc
-
-fromPlc :: UPLC.Program UPLC.NamedDeBruijn PLC.DefaultUni PLC.DefaultFun () -> Script
-fromPlc (UPLC.Program a v t) =
-    let nameless = UPLC.termMapNames UPLC.unNameDeBruijn t
-    in Script $ UPLC.Program a v nameless
+fromCompiledCode = Script . toNameless . getPlc
+    where
+      toNameless :: UPLC.Program UPLC.NamedDeBruijn PLC.DefaultUni PLC.DefaultFun ()
+                 -> UPLC.Program UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun ()
+      toNameless = over UPLC.progTerm $ UPLC.termMapNames UPLC.unNameDeBruijn
 
 data ScriptError =
     EvaluationError [Text] Haskell.String -- ^ Expected behavior of the engine (e.g. user-provided error)
     | EvaluationException Haskell.String Haskell.String -- ^ Unexpected behavior of the engine (a bug)
     deriving (Haskell.Show, Haskell.Eq, Generic, NFData)
 
-applyArguments :: Script -> [PLC.Data] -> Script
-applyArguments (Script (UPLC.Program a v t)) args =
+applyArguments :: (Wrapped s, Unwrapped s ~ UPLC.Program name PLC.DefaultUni PLC.DefaultFun ())
+               => s -> [PLC.Data] -> s
+applyArguments s args =
```

Comment:
@michaelpj  I think you are going to hate this abstracted applyarguments over wrapped. that was the only way i could share it with the Api.hs ,  w.o. code duplication

Reply:
I think we can just bin this function and inline it into `mkTermToEvaluate`. The functions in here that do stuff with `Script`s aren't really that useful.